nullaway/src/test/java/com/uber/nullaway/NullAwayCoreTests.java

```diff
@@ -990,4 +990,90 @@ public void annotationAppliedToTypeParameter() {
             "}")
         .doTest();
   }
+
+  @Test
+  public void annotationAppliedToInnerTypeImplicitly() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.checkerframework.checker.nullness.qual.Nullable;",
+            "class Test {",
+            "  @Nullable Foo f;", // i.e. Test.@Nullable Foo
+            "  class Foo { }",
+            "  public void test() {",
+            "    // BUG: Diagnostic contains: dereferenced",
+            "    f.hashCode();",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void annotationAppliedToInnerTypeExplicitly() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.checkerframework.checker.nullness.qual.Nullable;",
+            "class Test {",
+            "  Test.@Nullable Foo f1;",
+            "  @Nullable Test.Foo f2;",
```

Comment:
Is it a problem that we treat these two as strictly equivalent for now? Feels like we need to, in order to support the `annotationAppliedToInnerTypeImplicitly` case.

Reply:
I think we have to treat them equivalently for now.  The former is a correct usage of the annotation, and the latter we must allow for backward compatibility.  Do we know why Error Prone complains about the case below but not this one?